api/main.go

```diff
@@ -16,6 +16,21 @@ import (
 	"strings"
 )
 
+func withCORS(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*") // Allow all origins for CORS. Change it to specific origins in production.
```

Comment:
The allowed origin should come from "api/config/env.go", since it's much better to allow setting it up without the need to recompile from source. 

Once that's done, you'd need to check if the origin matches the whitelisted one.